leakcanary-android/src/main/java/com/squareup/leakcanary/internal/ForegroundService.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (C) 2018 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.leakcanary.internal;
+
+import android.app.IntentService;
+import android.app.Notification;
+import android.content.Intent;
+import android.os.IBinder;
+import android.support.annotation.Nullable;
+
+public abstract class ForegroundService extends IntentService {
+
+  private final int notificationContentTitleResId;
+
+  public ForegroundService(String name, int notificationContentTitleResId) {
```

Comment:
Constructors on a service may not always be used. Abstract method for the notification string?

Reply:
Sorry, I'm not sure I follow why the constructor might not be used (this is in the internal package, only meant for LeakCanary).

Reply:
An Intent triggers the service, meaning the OS is in charge of constructing it/restarting it. (Just like putting a constructor on an activity is bad)